### ISSUE TITLE
Self-training configuration and details

### ISSUE THREAD

**Issue by Cupcee** [valeoai/SALUDA#8]
Hey, this paper is very cool concept. We're trying to reproduce the results on our own dataset / segmentation task. We have a couple questions about the self-training step:

1. What choice for Tau (momentum) in EMA weight update did you use? CosMix used 0.9 and 0.999 depending on dataset.
2. How often did you update the EMA weights in your experiments: every batch, or once every N batches?
3. Did you threshold teacher model's pseudo-labels with some threshold, like the CosMix authors did?
4. How long (how many epochs) did you train the student?

Thank you.

**Comment by BjoernMichele** [comment]
Dear @Cupcee ,
 
Thank you for your interest in our work. Our self-training is very close to the CosMix training (of course without the mixing part) and we also built it on top of their code base and hyperparameter selection. 

If they provide the setting (like in Syn2SK or or Syn2Poss) we use the same Tau, update rate and number of epochs for training the student. For NS2SK, which they do not cover, we built up-on the Syn2SK setting but we did a 10 epochs self-training. However, normally the first epoch is gaining the most improvements, so 3 epochs should also be sufficient.  For our 10-cm voxel size experiments we thresholded the teacher pseudo-labels in all settings at 0.9.  


With best regards, 
Bjoern

**Comment by BjoernMichele** [comment]
Feel free to reopen the issue if there are any more questions.
